examples/audio/uk_ireland_accent_recognition.py

```diff
@@ -0,0 +1,664 @@
+
+"""
+### Train & Validation Sets
+
+Let's split the samples creating training and validation sets
+"""
+
+split = int(len(dataframe) * (1 - VALIDATION_RATIO))
+train_df = dataframe[:split]
+valid_df = dataframe[split:]
+
+print(
+    f"We have {train_df.shape[0]} training samples & {valid_df.shape[0]} validation ones"
+)
+
+"""
+## TensorFlow Dataset
+
+Next, we need to create a `tf.data` dataset.
+This is done by creating a `dataframe_to_dataset` function that does the following:
+*   Create a dataset using filenames and labels
+*   Get the Yamnet embeddings by calling another function `filepath_to_embeddings`
+*   Apply caching, reshuffling and setting batch size
+
+The `filepath_to_embeddings` does the following:
+*   Load audio file
+*   Resample audio to 16 kHz
+*   Generate scores and embeddings from Yamnet model
+*   Since Yamnet generates multiple samples for each audio file, this function also duplicates the label for all the generated samples that have `score = 0` i.e. speech whereas sets the label for the others as 'other' indicating that this audio segment is not a speech and we won't label it as one of the accents.
+
+"""
+
+"""
+The below **`load_16k_audio_file`** is copied from the following tutorial [Transfer
+learning with YAMNet for environmental sound
+classification](https://www.tensorflow.org/tutorials/audio/transfer_learning_audio)
+"""
+
+
+@tf.function
+def load_16k_audio_wav(filename):
+    # Read file content
+    file_content = tf.io.read_file(filename)
+
+    # Decode audio wave
+    audio_wav, sample_rate = tf.audio.decode_wav(file_content, desired_channels=1)
+    audio_wav = tf.squeeze(audio_wav, axis=-1)
+    sample_rate = tf.cast(sample_rate, dtype=tf.int64)
+
+    # Resample to 16k
+    audio_wav = tfio.audio.resample(audio_wav, rate_in=sample_rate, rate_out=16000)
+
+    return audio_wav
+
+
+def filepath_to_embeddings(filename, label):
+    # Load 16k audio wave
+    audio_wav = load_16k_audio_wav(filename)
+
+    # Get audio embeddings & scores.
+    # The embeddings are the audio features extracted using transfer learning
+    # while scores will be used to identify time slots that are not speech
+    # which will then be gathered into a specific new category 'other'
+    scores, embeddings, _ = yamnet_model(audio_wav)
+
+    # Number of embeddings in order to know how many times to repeat the label
+    embeddings_num = tf.shape(embeddings)[0]
+    labels = tf.repeat(label, embeddings_num)
+
+    # Change labels for time-slots that are not speech into a new category 'other'
+    labels = tf.where(tf.argmax(scores, axis=1) == 0, label, len(class_names) - 1)
+
+    # Using one-hot in order to use AUC
+    return (embeddings, tf.one_hot(labels, len(class_names)))
+
+
+def dataframe_to_dataset(dataframe, batch_size=64):
+    dataset = tf.data.Dataset.from_tensor_slices(
+        (dataframe["filename"], dataframe["label"])
+    )
+
+    dataset = dataset.map(lambda X, y: filepath_to_embeddings(X, y)).unbatch()
```

Comment:
Why `unbatch`?

Reply:
Because in filepath_to_embeddings function, for each filename, we get a different number of embeddings as output depending on the audio file length. so for a filename, we get (N1, 1024) as output and (N2, 1024) for a different input file. An error occurs when I batch them. That's why I unbatch them and then batch them to batch_size